pkg/master/controller.go

```diff
@@ -132,10 +132,23 @@ func (c *Controller) UpdateKubernetesService(reconcile bool) error {
 	return nil
 }
 
-// CreateNamespaceIfNeeded will create the namespace that contains the master services if it doesn't already exist
+// EnsureSystemNamespacesExist will make sure both the default and the system namespace always will exist
+func (c *Controller) EnsureSystemNamespacesExist() error {
+	// Try to create the default namespace
+	if err := c.CreateNamespaceIfNeeded(api.NamespaceDefault); err != nil {
+		return err
```

Comment:
also, should it attempt all namespaces, aggregating errors, rather than returning on the first failure?